Create restaurants.py

BreckoD opened the pull request:
Created a DataProvider for NYC Restaurant Inspection data.

BreckoD committed:
Create restaurants.py

Created a DataProvider for NYC Restaurant Inspection data.

yuribogomolov commented on src/sttn/data/restaurants.py:
```diff
@@ -0,0 +1,90 @@
+     where the inspection happened."""
+  
+    @staticmethod
+    def get_data(self, from_date, to_date) -> network.SpatioTemporalNetwork:
```
It's good practice to add the typing information for function arguments: `from_date: str`

yuribogomolov commented on src/sttn/data/restaurants.py:
```diff
@@ -0,0 +1,90 @@
+            Specifies the end date to get data from
+                
+
+        Returns:
```
Please add column-level documentation (see [this example](https://github.com/yuribogomolov/sttn/blob/main/src/sttn/data/nyc.py#L44))

yuribogomolov reviewed (approved):
Looks good to me! Please address the two comments below before merging

BreckoD committed:
Merge branch 'yuribogomolov:main' into patch-1

BreckoD committed:
Added columns description and typing information for function arguments in restaurant.py

BreckoD commented:
Addressed your comments in the last commit. Next time, I will add the new version with a different network structure to the nyc.py. So, the restaurants.py can be used while I'm working on the updated version.

BreckoD committed:
Added restaurants inspections DataProvider to nyc.py

Added the NYC Restaurants Inspections DataProvider, which now creates a bipartite graph where restaurants and types of inspections are nodes and edges represent inspection details. Also, added float64->float32 downcasting to reduce memory footprint and additional column filtering

BreckoD committed:
Delete src/sttn/data/restaurants.py

Deleted an old version of the restaurants network

BreckoD committed:
Merge branch 'yuribogomolov:main' into patch-1

BreckoD commented:
Now it should be 100% ready

BreckoD committed:
Merge branch 'yuribogomolov:main' into patch-1
